src/Razor/Microsoft.AspNetCore.Mvc.Razor.Extensions.Version2_X/src/ViewComponentTypeVisitor.cs

```diff
@@ -76,11 +75,12 @@ private static bool AttributeIsDefined(INamedTypeSymbol type, INamedTypeSymbol q
                 return false;
             }
 
-            var attribute = type.GetAttributes().Where(a => SymbolEqualityComparer.Default.Equals(a.AttributeClass, queryAttribute)).FirstOrDefault();
```

Comment:
Would replacing Where/FirstOrDefault with `Any` achieve the same optimization?

Reply:
Without other changes, no. However, if an extension method for `ImmutableArray<T>` existed similar to the one I created for `IReadOnlyList<T>` in #30446, it would work.

* It needs to be specialized for `ImmutableArray<T>` to avoid boxing the type and its enumerator
* It needs a generic argument to avoid capturing state

Reply:
I see, thanks for the explanation!

Reply:
Do you have a preference between the current approach and the extension method?